dpsim-models/src/MathUtils.cpp

```diff
@@ -97,6 +97,47 @@ Real Math::realFromVectorElement(const Matrix &mat, Matrix::Index row) {
   return mat(row, 0);
 }
 
+// TODO: [Georgii] resize existing matrix instead of creating a new one (performance, memory)
```

Comment:
Thanks for the hint. Do you mean to reuse it instead of `returnNonZeroElements` function?
`sparseView` removes zero or too small values and returns a `SparseMatrix` (not a `DenseMatrix`). 
The problem I see is that `SparseMatrix`  does not completely get rid of zero values, it stores the non-zero values along with their indeces. For example, if you substract 1.0 from a `SparseMatrix`, it will be substracted also from the zero-valued coefficients of the `SparseMatrix`.

And this is how `returnNonZeroElements` is used in `MNAEigenvalueExtractor:`

```
**mDiscreteEigenvalues = CPS::Math::returnNonZeroElements(discreteEigenvaluesIncludingZeros);
**mEigenvalues = 2.0 / mTimeStep * ((**mDiscreteEigenvalues).array() - 1.0) /
                   ((**mDiscreteEigenvalues).array() + 1.0);
```
In this equation I'd like that 1 is substracted only from non-zero-valued coefficients of `mDiscreteEigenvalues`